src/hooks/core/post/usePostHook.tsx

```diff
@@ -0,0 +1,129 @@
+            feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: false};
+          }
+        });
+        setUnFollow(data);
+      } else {
+        const newFeed = {...feedsContext?.feeds[indexFeed], is_following_target: false};
+        feedsContext.feeds[indexFeed] = newFeed;
+        unfollowUserAnon(dataFollowAnon);
+      }
+    } else if (!feedData?.anon_user_info_color_name) {
+      feedsContext?.feeds.forEach((feed, index) => {
+        if (feed?.actor?.id === user_id) {
+          feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: true};
+        }
+      });
+      setFollow(data);
+    } else {
+      const newFeed = {...feedsContext?.feeds[indexFeed], is_following_target: true};
+      feedsContext.feeds[indexFeed] = newFeed;
+      followUserAnon(dataFollowAnon);
+    }
+    setMainFeeds(feedsContext.feeds, feedsContextDispatch);
+  };
+
+  const followUnfollowTopic = async (item: any, topic: string, offset: number) => {
+    const id = removePrefixTopic(topic);
+    const feedTopic = feedsContext.topicFeeds
+      ? feedsContext.topicFeeds.filter((feed) => feed?.topics?.includes(id))
+      : [];
+    const feedMain = feedsContext.feeds;
+
+    const user_id = item?.actor?.id;
+    const username = item?.actor?.data?.username;
+    const data = {
+      user_id_follower: myProfile?.user_id,
+      user_id_followed: user_id,
+      username_follower: myProfile?.username,
+      username_followed: username,
+      follow_source: 'feed'
+    };
+    const dataFollowAnon = {
+      follow_source: 'post',
+      post_id: item?.id
+    };
+    const indexFeed = feedTopic?.findIndex((feed) => {
+      return feed?.id === item?.id;
+    });
+    const feedData = feedTopic[indexFeed];
+    if (feedData?.is_following_target) {
+      if (!feedData?.anon_user_info_color_name) {
+        feedTopic.forEach((feed, index) => {
+          if (feed?.actor?.id === user_id) {
+            feedTopic[index] = {...feedTopic[index], is_following_target: false};
+          }
+        });
+        feedMain.forEach((feed, index) => {
+          if (feed?.actor?.id === user_id) {
+            feedMain[index] = {...feedMain[index], is_following_target: false};
+          }
+        });
+        setUnFollow(data);
+      } else {
+        feedTopic[indexFeed] = {...feedTopic[indexFeed], is_following_target: false};
+        feedMain[indexFeed] = {...feedMain[indexFeed], is_following_target: false};
+        unfollowUserAnon(dataFollowAnon);
+      }
+    } else if (!feedData?.anon_user_info_color_name) {
+      feedTopic.forEach((feed, index) => {
+        if (feed?.actor?.id === user_id) {
+          feedTopic[index] = {...feedTopic[index], is_following_target: true};
+        }
+      });
+      feedMain.forEach((feed, index) => {
+        if (feed?.actor?.id === user_id) {
+          feedMain[index] = {...feedMain[index], is_following_target: true};
+        }
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

You are directly mutating the state here which is not recommended in React. Instead, you should create a new copy of the state, modify it, and then set the state with the new copy.
```diff
- feedsContext?.feeds.forEach((feed, index) => {
-   if (feed?.actor?.id === user_id) {
-     feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: false};
-   }
- });
+ const updatedFeeds = feedsContext?.feeds.map((feed) => {
+   if (feed?.actor?.id === user_id) {
+     return {...feed, is_following_target: false};
+   }
+   return feed;
+ });
+ setMainFeeds(updatedFeeds, feedsContextDispatch);
```
Apply similar changes to other places where you are directly mutating the state.

56, 120:
It seems like you're setting the entire feeds array again after each follow/unfollow operation. This could lead to unnecessary re-renders and performance issues if the feeds array is large. Consider optimizing this by only updating the specific feed that was followed/unfollowed.



<!-- This is an auto-generated comment by OSS CodeRabbit -->